src/Microsoft.DotNet.Darc/Darc/Operations/VirtualMonoRepo/CherryPickOperation.cs

```diff
@@ -0,0 +1,201 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.DotNet.Darc.Options.VirtualMonoRepo;
+using Microsoft.DotNet.DarcLib;
+using Microsoft.DotNet.DarcLib.Helpers;
+using Microsoft.DotNet.DarcLib.VirtualMonoRepo;
+using Microsoft.Extensions.Logging;
+
+#nullable enable
+namespace Microsoft.DotNet.Darc.Operations.VirtualMonoRepo;
+
+internal class CherryPickOperation : Operation
+{
+    private readonly CherryPickCommandLineOptions _options;
+    private readonly IProcessManager _processManager;
+    private readonly IFileSystem _fileSystem;
+    private readonly IVmrPatchHandler _patchHandler;
+    private readonly IVersionDetailsParser _versionDetailsParser;
+    private readonly ILocalGitRepoFactory _localGitRepoFactory;
+    private readonly IVmrInfo _vmrInfo;
+    private readonly ILogger<CherryPickOperation> _logger;
+
+    public CherryPickOperation(
+        CherryPickCommandLineOptions options,
+        IProcessManager processManager,
+        IFileSystem fileSystem,
+        IVmrPatchHandler patchHandler,
+        IVersionDetailsParser versionDetailsParser,
+        ILocalGitRepoFactory localGitRepoFactory,
+        IVmrInfo vmrInfo,
+        ILogger<CherryPickOperation> logger)
+    {
+        _options = options;
+        _processManager = processManager;
+        _fileSystem = fileSystem;
+        _patchHandler = patchHandler;
+        _versionDetailsParser = versionDetailsParser;
+        _localGitRepoFactory = localGitRepoFactory;
+        _vmrInfo = vmrInfo;
+        _logger = logger;
+    }
+
+    public override async Task<int> ExecuteAsync()
+    {
+        try
+        {
+            return await ExecuteInternalAsync();
+        }
+        catch (Exception e)
+        {
+            _logger.LogError("Cherry-pick operation failed: {message}", e.Message);
+            _logger.LogDebug("{exception}", e);
+            return Constants.ErrorCode;
+        }
+    }
+
+    private async Task<int> ExecuteInternalAsync()
+    {
+        // Step 1: Determine if we're in VMR or repo based on source-manifest.json existence
+        var currentDirectory = Environment.CurrentDirectory;
+        var gitRoot = _processManager.FindGitRoot(currentDirectory);
+        var sourceManifestPath = Path.Combine(gitRoot, VmrInfo.DefaultRelativeSourceManifestPath.Path);
```

Comment:
You can make `gitRoot` a `NativePath` and then you can combine paths with the overloaded `/` operator like below:

```suggestion
        var gitRoot = _processManager.FindGitRoot(currentDirectory);
        var sourceManifestPath = Path.Combine(gitRoot, VmrInfo.DefaultRelativeSourceManifestPath.Path);
```
```suggestion
        var gitRoot = new NativePath(_processManager.FindGitRoot(currentDirectory));
        var sourceManifestPath = gitRoot / VmrInfo.DefaultRelativeSourceManifestPath.Path;
```

Please use `NativePath` wherever you can in the newly added files.